### ISSUE TITLE
LLaMA's Loss Function Is Lost

### ISSUE THREAD

**Issue by maxsu** [facebookresearch/llama#133]
:Mandatory Loss Reference:

Hi, 

I can't find the training loop or objective function in the code base. Have I missed it, or is it... lost? 😳

The paper does mention training perplexity as a stand-in for training loss, for instance:

>    On most benchmarks, the performance improves steadily, and correlates with the training perplexity of the model

If the loss function is a standard perplexity or cross entropy metric, can you please link us to more information? 
If training loss is compiled from the model using standard transformer techniques, can you please comment on that?

Thanks!

**Comment by HuangOwen** [comment]
You can check the huggingface implementation, where the loss function is provided.
https://github.com/huggingface/transformers/blob/main/src/transformers/models/llama/modeling_llama.py

**Comment by GearLeeess-JOE** [comment]
> You can check the huggingface implementation, where the loss function is provided. https://github.com/huggingface/transformers/blob/main/src/transformers/models/llama/modeling_llama.py

Cool, I found this related line: https://github.com/huggingface/transformers/blob/892399c5ffdc3c7d0e4f9b2c1c81511293c1581b/src/transformers/models/llama/modeling_llama.py#L712

**Comment by jspisak** [comment]
closing but tagging @AurRod for viz..
